### PR TITLE
ISPN-6242 Migrate to CDI 1.2 with annotation bean discovery

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-6242

### CODE REVIEW

**Comment by slaskawi** [comment]
Hey @mkouba ! May I ask you for review?

File: integrationtests/as-integration-embedded/src/test/resources/beans.xml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+       version="1.2" bean-discovery-mode="all">
 
-   <interceptors>
-      <class>org.infinispan.jcache.annotation.InjectedCacheResultInterceptor</class>
-      <class>org.infinispan.jcache.annotation.InjectedCachePutInterceptor</class>
-      <class>org.infinispan.jcache.annotation.InjectedCacheRemoveEntryInterceptor</class>
-      <class>org.infinispan.jcache.annotation.InjectedCacheRemoveAllInterceptor</class>
-   </interceptors>
+    <interceptors>
+        <class>org.infinispan.jcache.annotation.InjectedCacheResultInterceptor</class>
```
Comment (mkouba) [inline integrationtests/as-integration-embedded/src/test/resources/beans.xml]: Have you considered enabling these interceptors globally (with `@Priority`)? I don't know the use case but it usually makes sense and simplifies usage.

**Comment by mkouba** [comment]
Looks good. But honestly I don't remember the details of `AdvancedCacheProducer`, `InfinispanExtensionEmbedded`, etc... So I cannot say anything about the "business logic".

File: integrationtests/as-integration-embedded/src/test/resources/beans.xml (hunk 1 above)
Comment (slaskawi) [inline integrationtests/as-integration-embedded/src/test/resources/beans.xml]: Thanks for the hint! I'll experiment with it.

Comment (slaskawi) [inline integrationtests/as-integration-embedded/src/test/resources/beans.xml]: Unfortunately it looks like those Interceptors can not be enabled by default. Basically there are 2 groups of them (`injected*` and standard ones). Those injected should be used when deploying an app on Application Server and they prevent JCache from closing resources it does not own (e.g. WF `CacheManager`s).

Here is a reference to the fix: https://github.com/infinispan/infinispan/pull/2232

**Comment by slaskawi** [comment]
Thanks for review @mkouba !

**Comment by tristantarrant** [comment]
Pushed to master, thanks

**Pull request closed without merging**